internal/report/store.go

```diff
@@ -89,23 +89,62 @@ func (s *PolicyReportStore) AddClusterPolicyReport(report *ClusterPolicyReport)
 	return nil
 }
 
+func (s *PolicyReportStore) lookForNamespacedReport(namespace string) (PolicyReport, bool) {
+	s.namespacedPolicyReportsMutex.Lock()
+	defer s.namespacedPolicyReportsMutex.Unlock()
+	if report, ok := s.namespacedPolicyReports[namespace]; ok {
+		return report, true
+	}
+	return PolicyReport{}, false
+}
+
 // Get PolicyReport by namespace
 func (s *PolicyReportStore) GetPolicyReport(namespace string) (PolicyReport, error) {
-	s.namespacedPolicyReportsMutex.RLock()
-	defer s.namespacedPolicyReportsMutex.RUnlock()
-	report, present := s.namespacedPolicyReports[namespace]
-	if present {
+	// check if we already have a report to the given namespace in the cache
+	if report, ok := s.lookForNamespacedReport(namespace); ok {
 		return report, nil
 	}
-	return PolicyReport{}, errors.New("report not found")
+	// nothing in the cache. Let's request it in the cluster
```

Comment:
The Kubernetes Go client already keeps its own cache, and uses watches to keep it up to date. Do we need to have our own cache for the PolicyReports? I would always try to get the PolicyReport by querying the client. In some cases this is going to use its own cache, when needed it will will hit the API server.

Reply:
Sorry folks, I agree with you. I don't know why I've kept the old in memory map . In my mind I though that we would like to keep that for some reason that I cannot remember. Any way, as we are using etcd as the backend, it does not make sense to have it if we can send data directly to the backend.